src/systems/filecoin_blockchain/struct/block/block.id

```diff
@@ -1,40 +1,39 @@
-import ipld "github.com/filecoin-project/specs/libraries/ipld"
 import filcrypto "github.com/filecoin-project/specs/algorithms/crypto"
 
 import st "github.com/filecoin-project/specs/systems/filecoin_vm/state_tree"
 import clock "github.com/filecoin-project/specs/systems/filecoin_nodes/clock"
 import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
 import msg "github.com/filecoin-project/specs/systems/filecoin_vm/message"
 
-type BlockCID ipld.CID
-type MessageRoot ipld.CID
-type ReceiptRoot ipld.CID
 type ChainWeight UVarint
 type ChainEpoch UVarint
+type MessageReceipt util.Bytes
 
 // On-chain representation of a block header.
 type BlockHeader struct {
     // Chain linking
-    Parents          Tipset
-    Weight           ChainWeight
-    Epoch            ChainEpoch
-
-    // Miner Info
-    MinerAddress     addr.Address
-
+    Parents                [&BlockHeader]
+    ParentWeight           ChainWeight
     // State
-    StateTree        st.StateTree
-    Messages         MessageRoot
-    MessageReceipts  ReceiptRoot
+    ParentStateTree        &st.StateTree
+    ParentMessageReceipts  &[&MessageReceipt]  // array-mapped trie ref
 
     // Consensus things
-    Timestamp        clock.UnixTime
+    Epoch                  ChainEpoch
+    Timestamp              clock.UnixTime
     Ticket
     ElectionProof
 
+    MinerAddress           addr.Address
+    ElectionPoSt           OnChainPoStVerifyInfo
+
+    // Proposed update
+    BLSAggregateSig        filcrypto.Signature
+    BLSMessageCIDs         &[&msg.UnsignedMessage]  // array-mapped trie
+    SECPMessageCIDs        &[&msg.SignedMessage]  // array-mapped trie
```

Comment:
Blocking: The `TxMeta` structure from the data_structures file exists in both implementations and was probably put there for a reason, probably header size reduction. Please obtain @whyrusleeping's approval before removing it (or just put it back).